lib/concurrent-ruby/concurrent/atomic/lock_local_var.rb

```diff
@@ -0,0 +1,20 @@
+require_relative 'fiber_local_var'
+require_relative 'thread_local_var'
+
+module Concurrent
+  def self.mutex_owned_per_thread?
+    mutex = Mutex.new
+
+    # Lock the mutex:
+    mutex.synchronize do
+      # Check if the mutex is still owned in a child fiber:
+      Fiber.new{mutex.owned?}.resume
+    end
+  end
```

Comment:
I'm thinking we should cache that and only do it once, and just return true on truffleruby/jruby because that's been always true for them (and creating a Fiber there is expensive)

Reply:
Or just do that logic at the top-level and declare a constant instead/in addition of the method

Reply:
What about making the method private because it’s only used once and probably shouldn’t be a public interface anyway?

Reply:
Okay, I marked it as private.

Reply:
Also, it's already only used once, and the result is already cached (which constant is defined) so I'm not sure what you want to change?

Reply:
Maybe we can merge this PR and then follow up with a JRuby/TruffleRuby specific changes as makes sense since I don't really know what the code should look like (nor do I think we should hard code anything if it might change in the future).